utils/helpers.py

```
@@ -0,0 +1,190 @@
+"""
+Helper utility functions for the self-correcting MCP runtime
+"""
+
+import json
+import time
+import hashlib
+from typing import Any, Dict, List, Optional, Union
```

Ruff (0.11.9)

`typing.Dict` is deprecated, use `dict` instead

(UP035)